src/shared/fs/templateRegistry.ts

```diff
@@ -108,31 +102,37 @@ export class AsyncCloudFormationTemplateRegistry {
             return this.instance
         }
 
-        // Show user a message indicating setup is in progress
-        if (this.setupProgressMessage === undefined) {
-            this.setupProgressMessage = vscode.window.withProgress(
-                {
-                    location: vscode.ProgressLocation.Notification,
-                    title: localize(
-                        'AWS.codelens.waitingForTemplateRegistry',
-                        'Scanning CloudFormation templates... (except paths configured in [search.exclude](command:workbench.action.openSettings?"@id:search.exclude"))'
-                    ),
-                    cancellable: true,
-                },
-                async (progress, token) => {
-                    token.onCancellationRequested(() => {
-                        // Allows for new message to be created if templateRegistry variable attempted to be used again
-                        this.setupProgressMessage = undefined
-                    })
-                    getLogger().debug('cfn: getInstance() requested, still initializing')
-                    while (!this.isSetup) {
-                        await sleep(2000)
+        const config = SamCliSettings.instance
+        if (config.get('enableCodeLenses', false)) {
```

Comment:
does sam debugging work with `enableCodeLenses=false` ?

sam debugging depends on the registry to resolve stuff: https://github.com/aws/aws-toolkit-vscode/blob/1db7975286b61ed462e782d7dfc2573a53c615df/src/shared/sam/debugger/awsSamDebugger.ts#L257-L259

Reply:
Will take a look. If we're getting the instance, it should (it'll build the registry on demand).

Reply:
Looks like a no. Will take a look.

Reply:
At worst, perhaps remove the `config.get('enableCodeLenses')` check for now. Then we need see what's calling `CloudFormationTemplateRegistry.instance` and try to minimize those calls.

Reply:
This triggered a larger, but necessary change: essentially, `globals.templateRegistry` needed to be async to ensure the registry could build.